core/Types.h

```diff
@@ -821,6 +821,7 @@ struct DispatchArgs {
 struct DispatchComponent {
     TypePtr receiver;
     MethodRef method;
+    u2 totalArgs;
```

Comment:
It's very strange to me to see this data structure changing given the PR description.

Could we instead do this change by changing the type of `lsp::SendResponse`?

A large part of my skepticism is seeing how much of calls.cc (which is not related to LSP or completion) has to change in service of this PR.

Reply:
Yeah a lot of the changes are simply there to thread through the `totalArgs` value but I can totally do this with `lsp::SendResponse` instead.